src/modules/credits/CreditsClient.ts

```diff
@@ -22,29 +22,105 @@ export class CreditsClient extends BaseClient {
     }
   }
 
+  /**
+   * Creates an SSE connection with reconnection logic
+   * @param address - The user's address
+   * @param onMessage - Callback function for received messages
+   * @param onError - Callback function for errors
+   * @returns EventSource with auto-reconnection capabilities
+   */
   createSSEConnection(
     address: string,
     onMessage: (data: CreditsResponse) => void,
     onError: (error: Event) => void
   ): EventSource {
-    const eventSource = new EventSource(
-      `${this.url}/users/${address}/credits/stream`
-    )
-
-    eventSource.onmessage = event => {
-      try {
-        const data = JSON.parse(event.data) as CreditsResponse
-        onMessage(data)
-      } catch (error) {
-        console.error('Error parsing SSE event data:', error)
+    let eventSource: EventSource | null = null
+    let retryCount = 0
+    const MAX_RETRIES = 5
+    const RETRY_DELAY_MS = 3000
+
+    // Create a wrapper object that appears as an EventSource to TypeScript
+    const wrapper = {} as EventSource
+
+    const connect = () => {
+      if (eventSource) {
+        // Close existing connection before creating a new one
+        eventSource.close()
+      }
+
+      eventSource = new EventSource(
+        `${this.url}/users/${address}/credits/stream`
+      )
+
+      // Copy all properties from the real EventSource to our wrapper
+      Object.defineProperties(
+        wrapper,
+        Object.getOwnPropertyDescriptors(eventSource)
+      )
```

Comment:
This seems to be doing a shallow copy of eventSource, but it might not be doing what you're expecting, which is to copy the functionality from the EventSource, as it's prototyped and this new object is not linked to the EventSource object.

I've tried doing:
```suggestion
Object.getOwnPropertyDescriptors(new EventSource("https://credits.decentraland.org/users/0xaca5bc79b0cd51b726d2eadfc747f7ad4dfe7efb/credits/stream"))
```
And I got as a response a empty object.
You might want to instantiate it, cast it and modify it or take another path.

Reply:
good catch, approach changed